File: csrc/263.cpp
#include <iostream>
#include <math.h>

using namespace std;

class Solution {
public:
    bool isUgly(int num) {
        if (num < 1) return false;
        while (num % 2 == 0 || num % 3 == 0 || num % 5 == 0) {
            if (num % 2 == 0) num = num / 2;
            if (num % 3 == 0) num = num / 3;
            if (num % 5 == 0) num = num / 5;
        }
        return num == 1;
    }
};

int main() {
    Solution solution;
    cout << solution.isUgly(6) << endl;
    cout << solution.isUgly(8) << endl;
    cout << solution.isUgly(14) << endl;
    return 0;
}

File: pysrc/263.py
class Solution:
    def isUgly(self, num: int) -> bool:
        if num < 1:
            return False
        while num > 1 and num % 2 == 0:
            num = num // 2
        while num > 1 and num % 3 == 0:
            num = num // 3
        while num > 1 and num % 5 == 0:
            num = num // 5
        return num == 1


if __name__ == '__main__':
    solution = Solution()
    print(solution.isUgly(6))
    print(solution.isUgly(8))
    print(solution.isUgly(14))
